### PR TITLE
Unittests transfers

### PR DESCRIPTION
**Pull request by bobloblaws93**
Added unit tests for the following classes:
-Transfers
-manage
-walletheader

### CODE REVIEW

File: src/pages/home/home.spec.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    }).compileComponents();
+
+  }));
+
+  beforeEach(() => {
+
+    fixture = TestBed.createComponent(HomePage);
+    comp    = fixture.componentInstance;
+
+  });
```
Comment (JTalarico) [inline src/pages/home/home.spec.ts]: Spacing! Remove whitespace 76 and 79

File: src/pages/home/wallet-header/wallet-header.component.spec.ts [hunk 2]
```diff
@@ -7,15 +7,36 @@ import { AuthConfig, AuthHttp } from "angular2-jwt";
...
+    getCurrency(code:any){
+      return Observable.of();
+    }
+    init(data: any) {
+      return Observable.of();
+    }
+    mapCurrencies(data:any){
+      return Observable.of();
+
+    }
```
Comment (JTalarico) [inline src/pages/home/wallet-header/wallet-header.component.spec.ts]: Remove space.

File: src/pages/otp/otp.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class OtpPage {
    */
   public requestCode() {
     if (this.request.valid) {
-      this.loading = true;
+      // this.loading = true;
```
Comment (JTalarico) [inline src/pages/otp/otp.ts]: What is this? Remove the comment

File: src/pages/transfers/transfers.spec.ts [hunk 4]
```diff
@@ -1,36 +1,57 @@
...
+import {AuthConfig, AuthHttp} from "angular2-jwt";
+import {AlertController, LoadingController, ModalController, NavController, NavParams,ViewController,IonicModule} from "ionic-angular";
+import {MyApp} from "../../app/app.component";
+import {AuthService} from "../../providers/auth.service";
+import {BankTransferService} from "../../providers/bank.service";
+import {TransferService} from "../../providers/transfer.service";
+import {Observable} from 'rxjs/Rx';
+import {WalletsService} from "../../providers/wallet.service";
+import {TransfersPage} from "./transfers";
+
```
Comment (JTalarico) [inline src/pages/transfers/transfers.spec.ts]: Spacing!

File: src/pages/transfers/transfers.spec.ts [hunk 5]
```diff
@@ -1,36 +1,57 @@
...
+  @Injectable()
+  class MockTransferService {
+    getTransferByToken(transferToken:any) {
+      return 1;
+    }
+    receive(data: any) {
+      return 2;    }
+    send(data:any){
+      return 1;
+
```
Comment (JTalarico) [inline src/pages/transfers/transfers.spec.ts]: Spacing!

File: src/pages/transfers/transfers.spec.ts [hunk 6]
```diff
@@ -67,22 +90,36 @@ describe("Transfer Component", () => {
...
-    expect(component.showTransferModal()).toBeDefined();
+
+  it("Transfer component should create", () => {
+      expect(component).toBeTruthy();
+  });
+
+  it("Test add transaction",() => {
+    expect(component.addTransaction).not.toBeNull();
+
+
```
Comment (JTalarico) [inline src/pages/transfers/transfers.spec.ts]: Spacing!

File: src/providers/contacts.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    it("contacts v2", async(inject([ContactsService], (service) => {
+      expect(service.getContacts()).toBeDefined();
+    })));
+
+  it("addContact should not be null", async(inject([ContactsService], (service) => {
+    expect(service.addContact()).toBeDefined();
+  })));
+
+
+
```
Comment (JTalarico) [inline src/providers/contacts.service.spec.ts]: So much spaces! Remove

File: src/providers/wallet.service.spec.ts [hunk 8]
```diff
@@ -42,4 +42,15 @@ describe("Wallet Service", () => {
...
 
+  it("update wallet response should defined", async(inject([WalletsService], (service) => {
+    var wallet =[{"id":1,"balance":1,"shown":3.00,"order":1}]
+    expect(service.updateWallet(wallet)).toBeDefined();
+  })));
+
+  it("update walletid response should defined", async(inject([WalletsService], (service) => {
+    expect(service.updateWalletId("1")).toBeDefined();
+  })));
+
```
Comment (JTalarico) [inline src/providers/wallet.service.spec.ts:53]: Spacing!

**Review by JTalarico — DISMISSED** [review]

File: src/providers/auth.service.ts [hunk 9]
```diff
@@ -96,7 +96,7 @@ export class AuthService {
           let msg = res.json();
           if (msg.errors) {
             msg = msg.errors;
-          }
+          }``
```
Comment (Scieon) [inline src/providers/auth.service.ts]: What is the `` for?

File: src/providers/transfer.service.ts [hunk 10]
```diff
@@ -14,7 +14,7 @@ export class TransferService {
   ) {}
 
   /**
-   * Sends a post request to the server to transfer money to another user.
+   * Sends a post request to the server to transfer money to another user.<<
```
Comment (Scieon) [inline src/providers/transfer.service.ts]: What is the << for?

**Review by Scieon — APPROVED** [review]
Good work improving the code coverage Ronnie. However, please divide your unit tests into smaller pull requests and never combine unit test and feature development in one request together.

**Pull request merged**